Sources/Rendering/LICOpenGL/SurfaceLICInterface/index.js

```diff
@@ -0,0 +1,727 @@
+// vtkLICInterface methods
+// ----------------------------------------------------------------------------
+function getQuadPoly(openGLRenderWindow) {
+  const quad = vtkHelper.newInstance();
+  quad.setOpenGLRenderWindow(openGLRenderWindow);
+  // build the CABO
+  const ptsArray = new Float32Array(12);
+  for (let i = 0; i < 4; i++) {
+    ptsArray[i * 3] = (i % 2) * 2 - 1.0;
+    ptsArray[i * 3 + 1] = i > 1 ? 1.0 : -1.0;
+    ptsArray[i * 3 + 2] = 0.0;
+  }
+
+  const tCoord = new Float32Array([0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 1.0]);
+
+  const cellArray = new Uint16Array(8);
+  cellArray[0] = 3;
+  cellArray[1] = 0;
+  cellArray[2] = 1;
+  cellArray[3] = 3;
+  cellArray[4] = 3;
+  cellArray[5] = 0;
+  cellArray[6] = 3;
+  cellArray[7] = 2;
+  const points = vtkDataArray.newInstance({
+    numberOfComponents: 3,
+    values: ptsArray,
+  });
+  points.setName('points');
+  const cells = vtkDataArray.newInstance({
+    numberOfComponents: 1,
+    values: cellArray,
+  });
+  const tArray = vtkDataArray.newInstance({
+    numberOfComponents: 2,
+    values: tCoord,
+  });
+
+  quad.getCABO().createVBO(cells, 'polys', Representation.SURFACE, {
+    points,
+    cellOffset: 0,
+    tcoords: tArray,
+  });
+  return quad;
+}
+
+function vtkSurfaceLICInterface(publicAPI, model) {
+  model.classHierarchy.push('vtkSurfaceLICInterface');
+
+  publicAPI.renderQuad = (bounds, program) => {
+    const poly = model.licQuad;
+    const gl = model.context;
+    let VAO = model.licQuadVAO;
+    if (!VAO) {
+      VAO = vtkVertexArrayObject.newInstance();
+      VAO.setOpenGLRenderWindow(model.openGLRenderWindow);
+      model.licQuadVAO = VAO;
+    }
+
+    if (model.previousProgramHash !== program.getMd5Hash()) {
+      VAO.shaderProgramChanged();
+      poly.getCABO().bind();
+      VAO.addAttributeArray(
+        program,
+        poly.getCABO(),
+        'vertexDC',
+        poly.getCABO().getVertexOffset(),
+        poly.getCABO().getStride(),
+        model.context.FLOAT,
+        3,
+        model.context.FALSE
+      );
+      VAO.addAttributeArray(
+        program,
+        poly.getCABO(),
+        'tcoordDC',
+        poly.getCABO().getTCoordOffset(),
+        poly.getCABO().getStride(),
+        model.context.FLOAT,
+        2,
+        model.context.FALSE
+      );
+      model.previousProgramHash = program.getMd5Hash();
+    }
+    gl.drawArrays(gl.TRIANGLES, 0, poly.getCABO().getElementCount());
+    VAO.release();
+  };
+
+  function generateGaussianNoise(
+    length,
+    numberOfNoiseLevels,
+    noiseImpulseProbability,
+    noiseImpulseBackgroundValue,
+    min,
+    max
+  ) {
+    const N = 2048;
+    const impulseProb = Math.max(0.0, Math.min(1.0, noiseImpulseProbability));
+    const noise = Float32Array.from({ length: length * length }, () => {
+      let val = 0;
+      if (impulseProb === 1.0 || Math.random() > 1.0 - impulseProb) {
+        for (let i = 0; i < N; ++i) {
+          val += Math.random();
```

Comment:
why not `N * Math.random()` ?

Reply:
Don't we need more than 1 sample to estimate a Gaussian distribution ?

Reply:
https://stackoverflow.com/questions/25582882/javascript-math-random-normal-distribution-gaussian-bell-curve

Reply:
That link says:

> Javascript Math.random is not a Normal Distribution

Reply:
yes, and it tells you how to generate a normal distribution from a uniform distribution.

Reply:
so should I use the Box-Muller transform ? This central limit estimation seemed sufficient to me.

Reply:
my bad, I did not notice it was CLT.
I think you can leave it as is. It would be great to keep the original (C++) comments. That helps to read the code :-)

I was also about to suggest to make things a bit more "JS" like but I think it is better to leave it close to the original code.